shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -230,35 +232,12 @@ public boolean sendKeyEvent(KeyEvent event) {
           updateEditingState();
           return true;
         } else if (selStart > 0) {
-          // Delete to the left/right of the cursor depending on direction of text.
-          // TODO(garyq): Explore how to obtain per-character direction. The
-          // isRTLCharAt() call below is returning blanket direction assumption
-          // based on the first character in the line.
-          boolean isRtl = mLayout.isRtlCharAt(mLayout.getLineForOffset(selStart));
-          try {
-            if (isRtl) {
-              Selection.extendRight(mEditable, mLayout);
-            } else {
-              Selection.extendLeft(mEditable, mLayout);
-            }
-          } catch (IndexOutOfBoundsException e) {
-            // On some Chinese devices (primarily Huawei, some Xiaomi),
-            // on initial app startup before focus is lost, the
-            // Selection.extendLeft and extendRight calls always extend
-            // from the index of the initial contents of mEditable. This
-            // try-catch will prevent crashing on Huawei devices by falling
-            // back to a simple way of deletion, although this a hack and
-            // will not handle emojis.
-            Selection.setSelection(mEditable, selStart, selStart - 1);
+          QwertyKeyListener qwertyKeyListener = new QwertyKeyListener(Capitalize.NONE, false);
```

Comment:
This object can be cached instead of re-created every time, though as usual it is a trade-off between speed and memory usage. Since keystrokes are manual rare events, caching this is optional. Your choice.

Reply:
I'm now using getInstance() which handles caching.